packages/block-renderer/src/components/pattern-renderer.tsx

```diff
@@ -33,17 +37,19 @@ const PatternRenderer = ( {
 		patternHtml = transformHtml( patternHtml );
 	}
 
-	let patternStyles = pattern?.styles ?? [];
+	let patternStyles = pattern?.styles ? [ ...styles, ...pattern.styles ] : [ ...styles ];
```

Comment:
IMO looks more simple, but up to you
```suggestion
	let patternStyles = [ ...styles, ...( pattern.styles || [] ) ];
```

Reply:
Also, according to the types - `pattern.styles` is `RenderedStyle[]`. Maybe we could avoid the condition? Or you have faced case when it's `undefined`?

Reply:
`pattern` can be undefined, which is why we have the type guard. I'm fine with the more compact `patternStyles` assignment, though